### PR TITLE
JS2-Week3 HW

### PR DESCRIPTION
**Pull request by Saikiruthiga**
Implemented below features
- added inputs for 2 player names and a button to start a quiz
- player name with score 0 will be displayed when click on start button

### CODE REVIEW

File: quizApp.js [hunk 1]
```diff
@@ -2,16 +2,105 @@ document.addEventListener("DOMContentLoaded", updateForm);
...
+    correctIcon2.id = "icon-wrong2";
+    player2Container.appendChild(correctIcon2);
+    const wrongIcon2 = document.createElement("i");
+    wrongIcon2.className = "fa fa-times-circle scorebtn-2";
+    wrongIcon2.id = "icon-wrong2";
+    player2Container.appendChild(wrongIcon2);
+    const numInput2 = document.createElement("input");
+    numInput2.type = "number";
+    numInput2.id = "num-input2";
+    player2Container.appendChild(numInput2);
```
Comment (islanderupnorth) [inline quizApp.js]: If you find yourself repeating the same code then you can repeat it by looping. In this case you can loop this section of code and using template literals change the `scorebtn-${index}` and other places you use the index.

File: quizApp.js [hunk 2]
```diff
@@ -2,16 +2,105 @@ document.addEventListener("DOMContentLoaded", updateForm);
...
+      addScore(player1Btn, score1, numInput1)
+    );
+    correctIcon2.addEventListener(
+      "click",
+      addScore(player2Btn, score2, numInput2)
+    );
+    wrongIcon1.addEventListener(
+      "click",
+      addScore(player2Btn, score2, numInput2)
+    );
```
Comment (islanderupnorth) [inline quizApp.js]: When looping over the above elements you can add the addEventListener's within the loop.

File: quizApp.js [hunk 3]
```diff
@@ -134,16 +232,16 @@ function checkAnswer(event, question) {
...
 function searchQuestion(event) {
+  //console.log(questionArray);
   const string = event.target.value.toLowerCase();
-  console.log(string);
   const filteredQuestion = questionArray.filter((item) => {
     const question = item.question.includes(string);
-    const options = item.options.some((option) => option.text.includes(string));
+    const options = item.options.some((item) => item.text.includes(string));
     const explanation = item.explanation.includes(string);
     return question || options || explanation;
```
Comment (islanderupnorth) [inline quizApp.js:225]: Nice use of displaying the values which have fallback values!

File: quizApp.js [hunk 4]
```diff
@@ -81,13 +170,15 @@ function onClickingShuffle(event) {
     }
   }
 }
-function displayQuestion(item) {
-  const section = document.getElementById("quiz-questions");
-  section.innerHTML = "";
+function displayQuestion(item, append = false) {
+  const section = document.getElementById("quizquestions");
```
Comment (islanderupnorth) [inline quizApp.js]: Your original `kebab-case` is more common practice in HTML

**Review by islanderupnorth — COMMENTED** [review]
Great work! It works as expected and really nice touch with the Audio being played at the end!

It's great how you have separated everything into their own functions. My major comments were to try implement DRY (Don't Repeat Yourself) principles and minimising writing the same code twice when calling the same functions. Otherwise, great logic!

File: quizApp.js (hunk 4 above)
Comment (Saikiruthiga) [reply quizApp.js]: Thank you for your feedback. I have implemented DRY principle and change the code into looping and followed kebab-case in html and css.

**Pull request merged**